docs/howto/build-guides/platforms.rst

```diff
@@ -0,0 +1,159 @@
+and will run on an ``amd64`` system with Ubuntu 24.04, use the
+``charmcraft.yaml`` snippet below:
+
+.. code-block:: yaml
+
+   base: ubuntu@24.04
+   platforms:
+     amd64:
+       build-on: [amd64]
+       build-for: [amd64]
+
+The ``build-on`` and ``build-for`` entries are identical, so the
+:ref:`shorthand notation <reference-platforms-shorthand>` can be used:
+
+.. code-block:: yaml
+
+   base: ubuntu@24.04
+   platforms:
+     amd64:
+
+The results are the same with either snippet. Building on ``amd64`` will
+produce one charm that runs on an ``amd64`` Ubuntu 24.04 system.
+
+.. note::
+
+  The OS of the build system doesn't affect what is built. Using LXD or
+  Multipass as a backend, Charmcraft will create a build environment with the
+  correct OS.
+
+Create a charm for all architectures
+------------------------------------
+
+For charms that are architecture-independent, use the following
+``charmcraft.yaml`` snippet to create a single charm that can run on any
+architecture:
+
+.. code-block:: yaml
+
+   base: ubuntu@24.04
+   platforms:
+     all:
+       build-on: [amd64]
+       build-for: [all]
+
+Building on ``amd64`` will produce one charm that runs on any architecture on
+an Ubuntu 24.04 system.
+
+The ``all`` keyword can only be used for the ``build-for`` keyword. If the
+charm can be built on multiple architectures, they must be added to the
+``build-on`` keyword:
+
+.. code-block:: yaml
+
+   base: ubuntu@24.04
+   platforms:
+     all:
+       build-on: [amd64, riscv64]
+       build-for: [all]
+
+Building on ``amd64`` or ``riscv64`` will produce one charm that runs on all
+architectures on Ubuntu 24.04 systems.
+
+
+Create a set of charms for multiple architectures
+-------------------------------------------------
+
+Charms may contain architecture-specific code. To build charms for specific
+architectures, use the following ``charmcraft.yaml`` snippet:
+
+.. code-block:: yaml
+
+   base: ubuntu@24.04
+   platforms:
+     amd64:
+       build-on: [amd64]
+       build-for: [amd64]
+     riscv64:
+       build-on: [riscv64]
+       build-for: [riscv64]
+
+Because the ``build-on`` and ``build-for`` entries are identical for each
+platform, the shorthand notation can be used instead:
+
+.. code-block:: yaml
+
+   base: ubuntu@24.04
+   platforms:
+     amd64:
+     riscv64:
+
+The results are the same with either snippet. Building on ``amd64`` will
+produce one charm that runs on ``amd64``. Building on ``riscv64`` will produce
+one charm that runs on ``riscv64``.
+
+
+Create a charm for a different architecture
+-------------------------------------------
+
+A charm may require cross-compilation to build. To create a charm for a
+different architecture, use the following ``charmcraft.yaml`` snippet:
+
+.. code-block:: yaml
+
+   base: ubuntu@24.04
+   platforms:
+     riscv64-cross:
+       build-on: [amd64]
+       build-for: [riscv64]
+
+Building on ``amd64`` will produce one charm that runs on ``riscv64``.
```

Comment:
Can you add a note here that the user is responsible for ensuring that their charm works on the other architecture?

All of the current plugins (`charm`, `python`, `poetry`, `uv`) will install python packages for the host architecture rather than the target architecture.

Reply:
I realised we don't have an issue for this: https://github.com/canonical/craft-parts/issues/974

Reply:
Added, do you have a preference for the name, "users" or "charm developers"?

Reply:
I think that here, "users" and "developers" are sufficiently distinct. That's what I'm using to signal between snap users and snap crafters.